astropy/units/quantity.py

```diff
@@ -286,6 +317,102 @@ class Quantity(np.ndarray):
 
     __array_priority__ = 10000
 
+    def __init_subclass__(cls, **kwargs):
+        super().__init_subclass__(**kwargs)
+
+        # Register the TypeVar for cached reuse.
+        # See __class_getitem__ for usage.
+        _quantitytype_registry[repr(cls)] = T.TypeVar(repr(cls), bound=cls)
+
+    if sys.version_info >= (3, 7):  # >= python 3.7  # TODO rm when bump min py
```

Comment:
I don't think it hurts for `__class_getitem__` to be present even on python 3.6, so maybe just add it directly?